index.js

```diff
@@ -422,6 +424,76 @@ module.exports = function math_plugin(md, options) {
         alt: ['paragraph', 'reference', 'blockquote', 'list']
     });
 
+    // Regex to capture any html prior to math block, the math block (single or multi line), and any html after the math block
+    const math_block_within_html_regex = /(?<html_before_math>[\s\S]*?)\$\$(?<math>[\s\S]+?)\$\$(?<html_after_math>(?:(?!\$\$[\s\S]+?\$\$)[\s\S])*)/gm;
+
+    // Regex to capture any html prior to math inline, the math inline (single line), and any html after the math inline
+    const math_inline_within_html_regex = /(?<html_before_math>[\s\S]*?)\$(?<math>.*?)\$(?<html_after_math>(?:(?!\$.*?\$)[\s\S])*)/gm;
+
+    md.core.ruler.push("math_block_in_html_block", (state) => {  
+        if (!enableMathBlockInHtml) {    
```

Comment:
If these are false, can we skip registering these rules instead?

Reply:
Sure.